pwndbg/aglib/regs.py

```diff
@@ -0,0 +1,296 @@
+"""
+Reading register value from the inferior, and provides a
+standardized interface to registers like "sp" and "pc".
+"""
+
+from __future__ import annotations
+
+import ctypes
+import re
+import sys
+from types import ModuleType
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Tuple
+from typing import cast
+
+import pwndbg
+import pwndbg.aglib.arch
+import pwndbg.aglib.proc
+import pwndbg.aglib.qemu
+import pwndbg.aglib.remote
+import pwndbg.aglib.typeinfo
+import pwndbg.lib.cache
+from pwndbg.dbg import EventType
+from pwndbg.lib.regs import BitFlags
+from pwndbg.lib.regs import RegisterSet
+from pwndbg.lib.regs import reg_sets
+
+
+@pwndbg.lib.cache.cache_until("stop", "prompt")
+def regs_in_frame(frame: pwndbg.dbg_mod.Frame) -> pwndbg.dbg_mod.Registers:
+    return frame.regs()
+
+
+@pwndbg.aglib.proc.OnlyWhenRunning
+def get_register(
+    name: str, frame: pwndbg.dbg_mod.Frame | None = None
+) -> pwndbg.dbg_mod.Value | None:
+    if frame is None:
+        frame = pwndbg.dbg.selected_frame()
+    assert (
+        frame is not None
+    ), "pwndbg.dbg.selected_frame() should never return None when marked with @OnlyWhenRunning"
+
+    regs = regs_in_frame(frame)
+
+    return regs.by_name(name) or regs.by_name(name.upper())
+
+
+@pwndbg.aglib.proc.OnlyWhenQemuKernel
+@pwndbg.aglib.proc.OnlyWhenRunning
+def get_qemu_register(name: str) -> int | None:
+    out = pwndbg.dbg.selected_inferior().send_monitor("info registers")
+    match = re.search(rf'{name.split("_")[0]}=\s+([\da-fA-F]+)\s+([\da-fA-F]+)', out)
+
+    if match:
+        base = int(match.group(1), 16)
+        limit = int(match.group(2), 16)
+
+        if name.endswith("LIMIT"):
+            return limit
+        else:
+            return base
```

Comment:
Random comment: we should probably add some comment here to explain wth is that LIMIT stuff